provider/cloudflare/cloudflare.go

```diff
@@ -271,6 +271,20 @@ func getCreateDNSRecordParam(cfc cloudFlareChange) cloudflare.CreateDNSRecordPar
 	}
 }
 
+func convertCloudflareError(err error) error {
+	var apiErr *cloudflare.Error
+	if errors.As(err, &apiErr) {
+		if apiErr.ClientRateLimited() || apiErr.StatusCode >= http.StatusInternalServerError {
+			// Handle rate limit error as a soft error
+			return provider.NewSoftError(err)
+		}
+	}
+	if strings.Contains(err.Error(), "exceeded available rate limit retries") {
```

Comment:
Maybe add a comment as a remainder that this is a workaround for an issue with the cloudflare client ?
So we know that we can remove this when fixed ?

Reply:
I added a comment